packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,54 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import { TransactionAnalysis, UTXO, FeeRate } from "./types";
+import { isRBFSignaled, calculateEffectiveFeeRate } from "./utils";
+
+/**
+ * Analyze a transaction to determine fee bumping options.
+ * @param transaction The transaction to analyze
+ * @param utxos The UTXOs spent in the transaction
+ * @param client The blockchain client
+ * @returns An analysis of the transaction's fee bumping options
+ */
+export async function analyzeTransaction(
+  transaction: Transaction,
+  utxos: UTXO[],
+  currentNetworkFeeRate: FeeRate
+): Promise<TransactionAnalysis> {
+  const canRBF = isRBFSignaled(transaction);
+  const canCPFP = transaction.outs.some((out) => out.value > 546); // 546 satoshis is typically considered dust
```

Comment:
should use the constants for this so we can consistently update. also might be good to make this parameterizable where an arg can be passed in for this value that defaults to the constant.

Reply:
But also actually, this isn't really definitive of whether or not you can CPFP. In fact, a dust output could be a good candidate for CPFP since it actually becomes useful since you'll be spending it right away. 

I would say what determines if a tx is CPFPable is if there is  an output that the wallet owns. With PSBT, there is the option where an output could include redeem script and bip32 derivation information. A naive use of this is that if this is included you assume this is a spendable CPFP output. If you wanted to be smarter, you could compare the descriptor/derivation for the output and the spending inputs OR the function could take an arg that describes a wallet config that would be used to spend. This is useful because you can actually CPFP someone else's transaction if one of the outputs belongs to you. This is trickier to figure out though since the raw tx doens't have any of this information, so if you have a tx sent to you that is pending you have to determine which output is yours. You could do this by taking a wallet config/descriptor and generate a bunch of addresses and then compare the tx outputs to see if any matches. That coudl give you false negatives though (imagine if the output goes to your 50th deposit address and you only check up to 40 addresses). Alternatively you could let the consumer code determine this and just tell the function/class which outputs are yours. OR you can generate a PSBT from the raw tx and add this information to the output (this would be done in the consumer code, like caravan/coordinator for example). 

This is definitely very tricky though, so apologies this comment is a little rambling.

Reply:
addressed while creating the new classes and buitling the transaction analyser class